src/Symfony/Component/Form/Tests/Extension/Core/Type/DateTypeTest.php

```diff
@@ -395,6 +395,54 @@ public function testSubmitFromInputRawDifferentPattern()
         $this->assertEquals('06*2010*02', $form->getViewData());
     }
 
+    public function testSubmitWithCalendarOption()
+    {
+        // we test against "en_US", so we need the full implementation
+        IntlTestHelper::requireFullIntl($this, false);
+
+        \Locale::setDefault('en_US');
+
+        $format = "y-MM-dd y'w'w";
+
+        $output = [
+            'day' => '31',
+            'month' => '3',
+            'year' => '2024',
+        ];
+
+        $form = $this->factory->create(static::TESTED_TYPE, null, [
+            'model_timezone' => 'UTC',
+            'view_timezone' => 'UTC',
+            'format' => $format,
+            'html5' => false,
+            'widget' => 'single_text',
+            'input' => 'array',
+        ]);
+
+        $form->submit('2024-03-31 2024w14');
+
+        $this->assertEquals($output, $form->getData());
+        $this->assertEquals('2024-03-31 2024w14', $form->getViewData());
+
+        // ISO 8601
+        $calendar = \IntlCalendar::createInstance();
+        $calendar->setFirstDayOfWeek(\IntlCalendar::DOW_MONDAY);
+        $calendar->setMinimalDaysInFirstWeek(4);
+
+        $form = $this->factory->create(static::TESTED_TYPE, null, [
+            'format' => $format,
+            'html5' => false,
+            'widget' => 'single_text',
+            'input' => 'array',
+            'calendar' => $calendar,
+        ]);
+
+        $form->submit('2024-03-31 2024w13');
+
+        $this->assertEquals($output, $form->getData());
+        $this->assertEquals('2024-03-31 2024w13', $form->getViewData());
```

Comment:
Can we use asserSame ?

Same for the other tests

Reply:
I have the change ready on local, but I'm not a big fan of it as other tests in changed files use `assertEquals`. Could somebody else give us an opinion?

Reply:
opinion: never rely on `assertEquals`, same as never rely on `==` instead of `===`

Reply:
You can change it to `assertSame` across the whole file 👍